From 3d81fe9f9c74e0dfc64fa62105879a1a3c7141b3 Mon Sep 17 00:00:00 2001
From: Johannes Wiesboeck <johannes.wiesboeck@aisec.fraunhofer.de>
Date: Tue, 5 Aug 2025 21:29:57 +0200
Subject: [PATCH] include/configs/tqmlx2160a.h: Set bootcommand

Set the bootcommand for booting gyroidos from sd card.
Secure boot for DPL and kernel.

Signed-off-by: Johannes Wiesboeck <johannes.wiesboeck@aisec.fraunhofer.de>
---
 include/configs/tqmlx2160a.h | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/include/configs/tqmlx2160a.h b/include/configs/tqmlx2160a.h
index 532afb79139..7bdffeb8898 100644
--- a/include/configs/tqmlx2160a.h
+++ b/include/configs/tqmlx2160a.h
@@ -344,7 +344,16 @@ unsigned long get_board_ddr_clk(void);
 #define XSPI_NOR_BOOTCOMMAND					\
 	"run distro_bootcmd;run xspi_bootcmd; "
 #define SD_BOOTCOMMAND						\
-	"run distro_bootcmd;run sd_bootcmd;"
+	"load mmc ${mmcblkdev}:${bootpart} 0x80d00000 ${dpl_file};"	\
+	"env exists mcinitcmd && env exists secureboot "        \
+	" && load mmc 0:1 0x806C0000 ${dpl_file}.hdr "              \
+	"&& esbc_validate 0x806C0000;env exists mcinitcmd && "     \
+	"fsl_mc lazyapply DPL 0x80d00000;" 			\
+	"load mmc ${mmcblkdev}:${bootpart} ${loadaddr} /fitImage-gyroidos-cml-initramfs;"	\
+	"env exists secureboot && " \
+	"load mmc ${mmcblkdev}:${bootpart} 0x80200000 /fitImage-gyroidos-cml-initramfs.hdr " \
+	" && esbc_validate 0x80200000; " \
+	"bootm ${loadaddr}"
 #define SD2_BOOTCOMMAND						\
 	"run distro_bootcmd;run emmc_bootcmd;"
 
